v-next/hardhat-utils/src/fs.ts

```diff
@@ -318,7 +321,7 @@ export async function writeJsonFileAsStream<T>(
 
     await pipeline(jsonStream, fileWriteStream);
   } catch (e) {
-    ensureError<NodeJS.ErrnoException>(e);
+    ensureError(e);
```

Comment:
same here

Reply:
This is also a mix of two types of errors.